---
title: "Get started with `RegionScan`"
author: "Myriam Brossard"
date: "December 02, 2024"
output:
   rmarkdown::pdf_document:
    fig_caption: yes
    includes:
      in_header: my_header.txt
vignette: >
  %/VignetteIndexEntry{vignette}
  %/VignetteEngine{knitr::rmarkdown}
  %/VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

`RegionScan` is a R package is designed for scalable genome-wide association testing of both region-level multiple-variant and single-variant statistics, with visualization of the results. For detection of association under heterogeneous regional architectures, it implements three classes of state-of-the-art region-level tests, including multiple-variant linear/logistic regression (with and without dimension reduction), variance-component score tests, and region-level minP tests. RegionScan also supports the analysis of multi-allelic variants and unbalanced binary phenotypes and is compatible with widely used variant call format (VCF) files for both genotyped and imputed variants. Association testing leverages linkage disequilibrium (LD) structure in pre-defined regions, for example, LD-adaptive regions obtained by genomic partitioning, and accommodates parallel processing to improve computational and memory efficiency.  Detailed outputs and utility functions are provided to assist comparison, visualization, and interpretation of results. These versatile features support a wide range of potential applications.

In this vignette, we illustrate basic usage of `RegionScan` functions applied to a reproducible example of dataset provided with the package. The list of main functions and options as well as description of the outputs can be found in Annex of this vignette.

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Installation

```{r setup, message=FALSE, warning=FALSE, results = 'hide'}
library(devtools)
install_github("brossardMyriam/RegionScan")
library(RegionScan)
```

## Basic usage of `regscan`

### Example dataset

This example dataset is based on 436 biallelic SNPs (MAF>0.05) genotyped in chr16:46382489-47684754 in 40,000 individuals simulated from high coverage whole genome sequenced 1000G European ancestry haplotypes using HAPGEN2 software. For this illustration, we used 20 consecutive regions identified by LD partitioning of chr16 using the BigLD algorithm (https://pubmed.ncbi.nlm.nih.gov/29028986/) implemented in R package *gpart* (https://academic.oup.com/bioinformatics/article/35/21/4419/5487391). However, *RegionScan* accepts any type of user-defined region boundaries (e.g. gene start/end positions etc). We simulated a quantitative trait (*sim_QT* in the `phenocov` input) and a binary trait (*sim_bin*) ; sim_QT was generated under a linear regression model, assuming joint effects of two causal SNPs in region 8 ("chr16.46880510.A.G", "chr16.46889594.G.A"); while *sim_bin* was generated by dichotomizing *sim_QT* (see PhenoSim.R for details). 

In this vignette, we illustrate how to run the main function *regscan* for region-level and single-SNP analysis and how to visualize the results.

### Main inputs

Three main inputs required by regscan

##### Input 1: REGIONinfo

This dataframe must include at least `chr`, `start.bp`, `end.bp`, `region`; regions start and end positions.

```{r,, warning=FALSE}
head(REGIONinfo)
```

##### Input 2: geno

This dataframe must include genotypes (columns) of the individuals (rows) ; as illustrated below for 4 genetic variants.
The individuals must be in the same order as the individuals in input `phenocov`

```{r,, warning=FALSE}
head(geno[,1:4])
```


##### Input 3: SNPinfo

This dataframe includes the variant positions and information for the variants in `geno` input ; all the following columns are required

```{r,, warning=FALSE}
head(SNPinfo)
```


##### Input 4: phenocov

This dataframe must includes phenotypes and covariates for all the individuals of input `geno`. The individuals must be in the same order as in input 'geno.

```{r,, warning=FALSE}
head(phenocov)
```

### Single & Region-level analysis
Example of run of `regscan` main function for region & single-SNP level analysis with the continuous outcome `sim_QT` . 
```{r, , warning=FALSE, message=FALSE, results = 'hide'}
results<-regscan(phenocov = phenocov, pheno="sim_QT",  REGIONinfo=REGIONinfo, 
                             geno_type="D", pheno_type="C", data = geno,  SNPinfo = SNPinfo )
```

### Main Outputs

The main output of `regscan` function is a list including the region-level, Bin-level, variant-level outputs and filtered variant lists; and optionally an additional output (sinleSNPall) with single-SNP results for all the variants analyzed (including filtered variants in region-level analysis).

##### Output 1: Region-level results

This is the main output which includes results from all the region-level test implemented in RegionScan for all regions from REGIONinfo.


```{r, , warning=FALSE}
head(results$regionout)
```

##### Output 2: Bin-level results

This output includes the bin-level association results (deltaB, deltaB.se, deltaB.pvalue) for each LD bin identified in the regions for the MLC test. It also includes the bin sizes (ie. number of SNPs in each LD bin before and after LD-based pruning. The bin-level results can help to investigate some MLC region-level test results.

```{r, , warning=FALSE}
head(results$binout)
```

##### Output 3: variant-level results

This output provides detailed results at the variant-level for all variants kept for the region-level analysis. The variant-specific results from multiple regression at the region level and single-SNP regression models are reported. Investigation of this output can help to investigate the region-level results (particularly for the MLC region-level test).

```{r, , warning=FALSE}
head(results$snpout)
```

##### Output 4: variants excluded

This output lists all the variants excluded within each region and the reasons of exclusion (MAF, LD pruning etc); the LD bin information is also reported.

```{r, , warning=FALSE}
head(results$filterout)
```

## Visualisation of the results

#### Locus plot

Illustration of the LocusPlot function which plots the region-level results in a set of consecutive regions. The plot is directly saved as a pdf in the local directory.

```{r, , warning=FALSE, message=FALSE}
results$regionout$chr<-as.numeric(as.character(results$regionout$chr))
LocusPlot(chr=16,pheno="sim_QT",regscanout=results,regionlist=c(1:15),outname="LocusPlot",
          region_tests=c("Wald.p","PC80.p","MLCB.p","SKATO.p"))

```

```{r pressure, echo=FALSE, fig.cap="Locus plot of consecutive regions", out.width = '100%'}
knitr::include_graphics("C:/Users/brossard/Documents/OneDrive_perso/OneDrive/Haplotypes_CLQD_MLC/RegionScan_paper/RegionScan_dev/example/LocusPlot_region1_to_15.pdf")
```

#### LD heatmaps

The following function produces LD heatmaps for a specified region (here region "8") to visualize the correlation structure within region, before and after pruning, as well as the dependencies between the LD bins. Plots are saved as pdf files in the local directory.

```{r, , warning=FALSE, message=FALSE, results = 'hide'}
regscan(phenocov = phenocov, pheno="sim_QT",  REGIONinfo=REGIONinfo,
        geno_type="D", pheno_type="C", data = geno,  SNPinfo = SNPinfo,
        MLCheatmap = TRUE, regionlist ="8" )
```

```{r pressure3, echo=FALSE, fig.cap="Example 1 of LD heatmap plot for region 8 (after LD pruning), ordered by variant positions", out.width = '100%'}
knitr::include_graphics("C:/Users/bross/OneDrive/Haplotypes_CLQD_MLC/RegionScan_paper/RegionScan_dev/GitHub2024_RegionScan/example/MLC_heatmap_chr16.region8_after_pruning_and_after_recoding_ordered_by_pos.pdf")
```

```{r pressure4, echo=FALSE, fig.cap="Example 2 of LD heatmap plot for region 8 (after LD pruning), ordered by LD bins", out.width = '100%'}
knitr::include_graphics("C:/Users/bross/OneDrive/Haplotypes_CLQD_MLC/RegionScan_paper/RegionScan_dev/GitHub2024_RegionScan/example/MLC_heatmap_chr16.region8_after_pruning_and_after_recoding_ordered_by_LDbin.pdf")
```

#### SNP LD bin positions

Visualization of the variant positions (X axis) along the LD bins (Y axis) for the variants kept after LD pruning (and in grey, removed by LD pruning). Plots are saved as pdf files in the local directory.

```{r, , warning=FALSE, message=FALSE, results = 'hide'}
MLCbinsnpPlot(rscanout = results, chr_=16, region_ =8 )
```

```{r pressure2, echo=FALSE, fig.cap="variant positions (X axis) along LD bins (Y axis) in region 8", out.width = '100%'}
knitr::include_graphics("C:/Users/bross/OneDrive/Haplotypes_CLQD_MLC/RegionScan_paper/RegionScan_dev/GitHub2024_RegionScan/example/MLC_LDbin_SNPpos_chr16_region8_before_and_after_pruning.pdf")
```


## Citation
RegionScan : A comprehensive R package for region-level genome-wide association testing with integration and visualization of multiple-variant and single-variant hypothesis testing.
Brossard M, Roshandel D, Luo K, Yavartanoo F, Paterson AD, Yoo YJ, Bull SB. https://www.biorxiv.org/content/10.1101/2024.03.04.582374v1.

## License
This package is released under the [GNU General Public License (GPL) v3.0.](https://www.gnu.org/licenses/gpl-3.0.html)

